data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/BulkRetryStrategy.java

```diff
@@ -238,13 +242,17 @@ private void handleRetry(final AccumulatingBulkRequest request, final BulkRespon
                     sentDocumentsOnFirstAttemptCounter.increment(numberOfDocs);
                 }
                 sentDocumentsCounter.increment(bulkRequestForRetry.getOperationsCount());
+                for (final BulkOperationWithHandle bulkOperation: bulkRequestForRetry.getOperations()) {
+                    EventHandle eventHandle = bulkOperation.getEventHandle();
+                    acknowledgementSetManager.releaseEventReference(eventHandle, true);
```

Comment:
I wonder if we should add a `releaseEventReference(boolean)` method directly to `EventHandle`. Then it can call the `AcknowledgementSetManager`. This could simplify this code quite a bit.

Reply:
I don't think that would be possible because each EventHandle won't have a reference to `AcknowledgementSetManager` unless we add `theInstance()` kind of API

Reply:
I think we should brainstorm this more. May be we can just do release by calling `AcknowledgementSet.release` instead of going through `AcknowledgementSetManager`